platforms/jvm/plugins-test-report-aggregation/src/main/java/org/gradle/api/plugins/TestReportAggregationPlugin.java

```diff
@@ -63,10 +62,18 @@ public abstract class TestReportAggregationPlugin implements Plugin<Project> {
     public void apply(Project project) {
         project.getPluginManager().apply("org.gradle.reporting-base");
 
-        ConfigurationContainer configurations = ((ProjectInternal) project).getConfigurations();
-        final Configuration testAggregation = configurations.dependencyScope(TEST_REPORT_AGGREGATION_CONFIGURATION_NAME).get();
-        testAggregation.setDescription("A configuration to collect test execution results");
-        testAggregation.setVisible(false);
+        ConfigurationContainer configurations = project.getConfigurations();
+        final Configuration testAggregation = configurations.dependencyScope(TEST_REPORT_AGGREGATION_CONFIGURATION_NAME, dependencyScope -> {
+            dependencyScope.setDescription("A configuration to collect test execution results.");
+            dependencyScope.setVisible(false);
+        }).get();
+
+        // A resolvable configuration to collect test results
+        Configuration testResultsConf = configurations.resolvable("aggregateTestReportResults", resolvable -> {
+            resolvable.extendsFrom(testAggregation);
+            resolvable.setDescription("Graph needed for the aggregated test results report.");
+            resolvable.setVisible(false);
```

Comment:
`visible = false` is default for the new role-based configuration factory methods